packages/browser/src/plugins/ajs-destination/index.ts

```diff
@@ -73,8 +74,8 @@ export class LegacyDestination implements DestinationPlugin {
 
   private _ready = false
   private _initialized = false
-  private onReady: Promise<unknown> | undefined
-  private onInitialize: Promise<unknown> | undefined
+  private readyPromise = createDeferred<void>()
```

Comment:
These are more like "wrappers around Promise" so maybe "readyDeferred" would be a better name? "deferred" is a common name for this object.

Reply:
`readyDeferred` doesn't really communicate the meaning of what's inside at a glance - `readyState`? then it'd be `readyState.promise`, `readyState.resolve()`

Reply:
Sounds good to me